src/main/groovy/org/mockito/release/internal/gradle/GitPlugin.java

```diff
@@ -3,78 +3,104 @@
+                    public void execute(Task task) {
+                        List<Object> arguments = new ArrayList<Object>();
+                        arguments.add("git");
+                        arguments.add("commit");
+                        arguments.add("--author");
+                        arguments.add(GitUtil.getGitGenericUserNotation(conf));
+                        arguments.add("-m");
+                        arguments.add(GitUtil.getCommitMessage(conf, t.getAggregatedCommitMessage()));
+                        arguments.addAll(t.getFiles());
+
+                        t.commandLine(arguments);
                     }
                 });
             }
         });
 
-        TaskMaker.execTask(project, TAG_TASK, new Action<Exec>() {
+        TaskMaker.execTask(project, GIT_TAG_TASK, new Action<Exec>() {
             public void execute(final Exec t) {
-                t.mustRunAfter(COMMIT_TASK);
+                t.mustRunAfter(GIT_COMMIT_TASK);
                 final String tag = GitUtil.getTag(conf, project);
                 t.setDescription("Creates new version tag '" + tag + "'");
                 deferredConfiguration(project, new Runnable() {
                     @Override
                     public void run() {
                         t.commandLine("git", "tag", "-a", tag, "-m",
-                        GitUtil.getCommitMessage(conf, "Created new tag " + tag));
+                                GitUtil.getCommitMessage(conf, "Created new tag " + tag));
                     }
                 });
             }
         });
 
-        TaskMaker.task(project, PUSH_TASK, SecureExecTask.class, new Action<SecureExecTask>() {
+        TaskMaker.task(project, GIT_PUSH_TASK, SecureExecTask.class, new Action<SecureExecTask>() {
             public void execute(final SecureExecTask t) {
-                t.setDescription("Pushes changes to remote repo.");
-                t.mustRunAfter(COMMIT_TASK, TAG_TASK);
+                t.setDescription("Pushes automatically created commits to remote repo.");
+                t.mustRunAfter(GIT_COMMIT_TASK);
+                t.mustRunAfter(GIT_TAG_TASK);
 
                 lazyConfiguration(t, new Runnable() {
                     public void run() {
-                        t.setCommandLine(GitUtil.getGitPushArgsWithTag(conf, project, gitStatus.getBranch()));
+                        t.setCommandLine(GitUtil.getGitPushArgs(conf, gitStatus.getBranch()));
                         t.setSecretValue(conf.getGitHub().getWriteAuthToken());
                     }
                 });
             }
         });
 
+        TaskMaker.task(project, PERFORM_GIT_PUSH_TASK, Task.class, new Action<Task>() {
+            public void execute(final Task t) {
+                t.setDescription("Pushes automatically created commits to remote repo.");
+                t.dependsOn(GIT_COMMIT_TASK, GIT_PUSH_TASK);
```

Comment:
I think we're missing dependsOn git tag here. This dependency was previously configured in the performRelease task. Without this dependency, git tag may run after git push and the build would break.